optuna/samplers/_partial_fixed.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any
+from typing import Dict
+import warnings
+
+import numpy
+
+from optuna._experimental import experimental
+from optuna.distributions import BaseDistribution
+from optuna.samplers import BaseSampler
+from optuna.study import Study
+from optuna.trial import FrozenTrial
+
+
+@experimental("2.3.0")
+class PartialFixedSampler(BaseSampler):
+    """Sampler that can sample parameters with fixed parameters.
+
+        .. versionadded:: 2.3.0
+
+    Example:
+
+         After several steps of optimization, you can fix the value of ``y`` and re-optimize it.
+
+        .. testcode::
+
+            import optuna
+
+
+            def objective(trial):
+                x = trial.suggest_uniform("x", -1, 1)
+                y = trial.suggest_int("y", -1, 1)
+                return x ** 2 + y
+
+
+            study = optuna.create_study()
+            study.optimize(objective, n_trials=10)
+
+            best_params = study.best_params
+            fixed_params = {"y": best_params["y"]}
+            partial_sampler = PartialFixedSampler(fixed_params, study.sampler)
```

Comment:
The current example code does not import `PartialFixedSampler`; you might need to use `optuna.samplers.PartialFixedSampler`.